### PR TITLE
getNodeByNodePublicKey api endpoint added

### LINKED ISSUE #95: Add endpoint that will allow get node by nodePublicKey
Node has two public keys (https://symbol.cryptobeliever.net:3001/node/info) publicKey and nodePublicKey:

{
"version": 16777984,
**"publicKey": "BB76775FDEBB42EFD081C7D277108E038420CB3BE7EC0CC89B81C624093D24D3",**
"networkGenerationHashSeed": "57F7DA205008026C776CB6AED843393F04CD458E0AA2D9F1D5F31A402072B2D6",
"roles": 3,
"port": 7900,
"networkIdentifier": 104,
"host": "symbol.cryptobeliever.net",
"friendlyName": "cryptobeliever.node (max slots 100) - 100% for harvester",
**"nodePublicKey": "D0B7F5FE7F067B4552FC6151BF0C9D9586126B132597E3401CC5843E4151DDF8"**
}

Currently statistics service API allow only to fetch node details by `publicKey`. We need also option to get by `nodePublicKey`. This will be used by wallets to determinate on which node account is harvesting (in node key link transactions we using nodePublicKey and node publicKey).

### PR DESCRIPTION
**Pull request by yilmazbahadir**
fixes #95

### CODE REVIEW

File: openapi/openapi.yml [hunk 1]
```diff
@@ -74,6 +74,33 @@ paths:
...
+                  example: E3FC28889BDE31406465167F1D9D6A16DCA1FF67A3BABFA5E5A8596478848F78
+            responses:
+                200:
+                    description: The node information for the given transport public key.
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/NodeInfo'
+                404:
+                    description: There is no node with the given transport public key.
```
Comment (rg911) [inline openapi/openapi.yml:99]: `no node info found ` better?

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "description": "",
     "scripts": {
         "dev": "nodemon",
-        "build": "rimraf ./dist/ && tsc",
+        "build": "rimraf ./dist/ && tsc && shx cp -r openapi dist/openapi",
```
[the remark this answers is not in the document]
Comment (yilmazbahadir) [reply package.json]: agree, removed.

File: openapi/openapi.yml [hunk 3]
```diff
@@ -74,6 +74,33 @@ paths:
...
+            tags:
+                - node
+            operationId: getNodeByNodePublicKey
+            parameters:
+                - name: nodePublicKey
+                  required: true
+                  description: The node's transport public key.
+                  in: path
+                  schema:
+                      type: string
```
[the remark this answers is not in the document]
Comment (yilmazbahadir) [reply openapi/openapi.yml:90]: pattern added for both public key parameters

File: openapi/openapi.yml (hunk 1 above)
Comment (yilmazbahadir) [reply openapi/openapi.yml:101]: I guess this is more like a description (copied from other API endpoint - publicKey).
Actual response is like below.
```
{
code: "ResourceNotFound",
message: "no resource exists with nodePublicKey 'DC5BBE8070876B35829C157E496BAD83C84F0DE22B9F489FC722C482F63A52F'"
}
```

File: src/routes/index.ts [hunk 4]
```diff
@@ -77,6 +77,20 @@ export class Routes {
 				.catch((error) => InternalServerError.send(res, error));
 		});
 
+		app.get('/nodes/nodePublicKey/:nodePublicKey', (req: Request, res: Response) => {
```
Comment (AnthonyLaw) [inline src/routes/index.ts:80]: do you think we should rename `nodePublicKey ` -> `transportPublicKey`?

I think we need change on Wallet and Explorer as well?

File: openapi/openapi.yml [hunk 5]
```diff
@@ -74,6 +75,34 @@ paths:
...
+                      pattern: '^([0-9A-Fa-f]){64}$'
+                  example: E3FC28889BDE31406465167F1D9D6A16DCA1FF67A3BABFA5E5A8596478848F78
+            responses:
+                200:
+                    description: The node information for the given transport public key.
```
Comment (AnthonyLaw) [inline openapi/openapi.yml:95]: depend what we want to use `nodePublicKey ` or `transportPublicKey`

`transport public key` -> `node public key`

File: src/routes/index.ts [hunk 6]
```diff
@@ -77,6 +77,20 @@ export class Routes {
...
+			const nodePublicKey = req.params.nodePublicKey;
+
+			return DataBase.getNodeByNodePublicKey(nodePublicKey)
+				.then((node) => {
+					if (node) {
+						res.send(node);
+					} else {
+						NotFoundError.send(res, 'nodePublicKey', nodePublicKey);
```
Comment (AnthonyLaw) [inline src/routes/index.ts:88]: Naming issue same as above.

**Review by AnthonyLaw — COMMENTED** [review]
Look good, the one issue about naming.
Symbol bootstrap used transport public key, I believe.
maybe we should rename `nodePublicKey -> transportPublicKey`

File: src/routes/index.ts (hunk 4 above)
Comment (OlegMakarenko) [reply src/routes/index.ts:80]: Catapult-REST returns it as `nodePublicKey`. IMO, it should be consistent.

Comment (OlegMakarenko) [reply src/routes/index.ts:80]: But if we decide to change the name, it should be also applied to Catapult-REST.

Comment (yilmazbahadir) [reply src/routes/index.ts:80]: >Catapult-REST returns it as nodePublicKey. IMO, it should be consistent.

Agree, that's why I kept the name.

**Review by OlegMakarenko — APPROVED** [review]

**Pull request merged**